test/test-packet60.js

```diff
@@ -0,0 +1,267 @@
+    CLIENT_PRV_KEY = require('fs').readFileSync(__dirname
+                                                + '/fixtures/id_rsa'),
+    PARSED_CLIENT_PUB_KEY = require('../lib/utils').parseKey(CLIENT_PUB_KEY),
+    PARSED_CLIENT_PRV_KEY = require('../lib/utils').parseKey(CLIENT_PRV_KEY);
+
+
+function makePair(cb) {
+  var server = new SSH2Stream({
+    server: true,
+    privateKey: SERVER_KEY
+  }), client = new SSH2Stream({});
+
+  var done = [];
+  function tryDone(who) {
+    done.push(who);
+    if (done.length != 2) return;
+    cb(server, client);
+  }
+
+  server.on('NEWKEYS', function () { tryDone('server'); });
+  client.on('NEWKEYS', function () { tryDone('client'); });
+  server.pipe(client).pipe(server);
+}
+
+function signWithClientKey(blob, syncCb) {
+  var signType = (PARSED_CLIENT_PRV_KEY.type ===
+      'rsa' ? 'RSA-SHA1' : 'DSA-SHA1');
+  var signature = crypto.createSign(signType);
+  signature.update(blob);
+  signature = signature.sign(PARSED_CLIENT_PRV_KEY.privateOrig, 'binary');
+  signature = new Buffer(signature, 'binary');
+
+  if (PARSED_CLIENT_PRV_KEY.type === 'dss' && signature.length > 40) {
+    // this is a quick and dirty way to get from DER encoded r and s that
+    // OpenSSL gives us, to just the bare values back to back (40 bytes
+    // total) like OpenSSH (and possibly others) are expecting
+    var newsig = new Buffer(40),
+        rlen = signature[3],
+        rstart = 4,
+        sstart = 4 + 1 + rlen + 1;
+    while (signature[rstart] === 0)
+      ++rstart;
+    while (signature[sstart] === 0)
+      ++sstart;
+    signature.copy(newsig, 0, rstart, rstart + 20);
+    signature.copy(newsig, 20, sstart, sstart + 20);
+    signature = newsig;
+  }
+  syncCb(signature);
+}
+
+function bufferEqual(a, b) {
+  if (a.length !== b.length) return false;
+  for (var i = 0; i < a.length; i++) {
+    if (a[i] !== b[i]) return false;
+  }
+  return true;
+}
+
+function publickey(server, client) {
+  server.on('USERAUTH_REQUEST', function (user, service, method, data) {
+    assert.equal(user, 'bob');
+    assert.equal(service, 'ssh-connection');
+    assert.equal(method, 'publickey');
+    assert.equal(data.keyAlgo, PARSED_CLIENT_PUB_KEY.fulltype);
+    assert.equal(true, bufferEqual(data.key, PARSED_CLIENT_PUB_KEY.public));
+    assert.equal(data.signature, undefined);
+    assert.equal(data.blob, undefined);
+    return server.authPKOK(data.keyAlgo, data.key);
+  });
+  client.on('USERAUTH_PK_OK', function () {
+    next();
+  }).authPK('bob', PARSED_CLIENT_PUB_KEY);
+}
+
+function keyboardInteractive(server, client) {
+  var infoReqsRxed = 0;
+
+  server.on('USERAUTH_REQUEST', function (user, service, method, data) {
+    assert.equal(user, 'bob');
+    assert.equal(service, 'ssh-connection');
+    assert.equal(method, 'keyboard-interactive');
+    assert.equal(data, '');
+    process.nextTick(function () {
+      server.authInfoReq('req 0', 'instructions',
```

Comment:
Style nit: try to keep arguments either all on the same line or split all of them (indented and lined up) for consistency, like:

```javascript
server.authInfoReq('req 0',
                   'instructions',
                   [ { prompt: 'Say something to req 0', echo: true } ]);
```